src/gameState.ts

```diff
@@ -7,25 +7,34 @@ export interface GameState {
 }
 
 export interface Cell {
-  value: "empty" | "black";
+  value: "empty" | "black" | "white";
 }
 
-export function playerHasWon(gameState: GameState): boolean {
-  const hexBoardGraph = createGraphFromGameState(gameState);
-  return doesPathExistForPlayer(hexBoardGraph);
-}
+export type stoneColor = "black" | "white";
 
-function doesPathExistForPlayer(hexBoardGraph: HexBoardGraph): boolean {
-  const pathBlack = doesPathExist(hexBoardGraph, "black-start", "black-end");
-  const pathWhite = doesPathExist(hexBoardGraph, "white-start", "white-end");
+export function playerHasWon(
+  gameState: GameState,
+  stoneColor: stoneColor
+): boolean {
+  const hexBoardGraph = createGraphFromGameState(gameState, stoneColor);
+  return doesPathExistForPlayer(hexBoardGraph, stoneColor);
+}
 
-  return !!(pathBlack || pathWhite);
+function doesPathExistForPlayer(
+  hexBoardGraph: HexBoardGraph,
+  stoneColor: stoneColor
+): boolean {
+  if (stoneColor == "black") {
+    return !!doesPathExist(hexBoardGraph, "black-start", "black-end");
+  } else {
+    return !!doesPathExist(hexBoardGraph, "white-start", "white-end");
+  }
```

Comment:
Actually I made a mistake here, the `doesPathExist()` method should return a boolean, hence, the `!!` syntax should be in this method rather than in the method `doesPathExistForPlayer()`. Can you move it please?